### PR TITLE
Configurable filesystem properties collected for Solaris2.x

### PR DESCRIPTION
**Pull request by mcquin**
Resolves https://github.com/chef/ohai/issues/629.

@chef/client-core

### CODE REVIEW

**Comment by btm** [comment]
@mcquin, How would this patch be used, would you put something like this in your client.rb?

```
ohai.plugin[:filesystem][:zfs_properties] = [ "used", "available" ]
```

It looks like the recently released ChefDK 0.9.0 is the first release with the plugin system from #613 (Ohai 8.6.0), so that's required for this patch.

@bdha, would this patch solve an issue for you?

File: lib/ohai/plugins/solaris2/filesystem.rb [hunk 1]
```diff
@@ -63,7 +63,15 @@
...
     # Grab any zfs data from "zfs get"
     zfs = Mash.new
-    so = shell_out("zfs get -p -H all")
+    # ohai.plugin[:filesystem][:zfs_properties] = 'used'
+    # ohai.plugin[:filesystem][:zfs_properties] = ['mountpoint', 'creation', 'available', 'used']
+    zfs_get = "zfs get -p -H "
+    if configuration(:zfs_properties).nil? || configuration(:zfs_properties).empty?
+      zfs_get << "all"
+    else
+      zfs_get << configuration(:zfs_properties).join(',')
```
Comment (btm) [inline lib/ohai/plugins/solaris2/filesystem.rb]: Is there magic elsewhere that would allow zfs_properties to be a string?

**Comment by mcquin** [comment]
[The comment here](https://github.com/chef/ohai/blob/mcquin/ohai-629/lib/ohai/plugins/solaris2/filesystem.rb#L66-L67) is a little non-obvious, but it demonstrates what to place in your client.rb

File: lib/ohai/plugins/solaris2/filesystem.rb (hunk 1 above)
Comment (mcquin) [inline lib/ohai/plugins/solaris2/filesystem.rb]: There is now ;)

**Comment by btm** [comment]
Great, which appears congruent with the example I came up with.

:+1:

**Comment by bdha** [comment]
Thanks for looking at this! I've been rolling my own Chef dists for a few years now and this is one of the big reasons why.

I'd argue for swapping what you collect -- be conservative first. Otherwise with the default behavior someone with a lot of snapshots is still going to have a bad experience. `chef-client` will take an excessively long time to run, or simply never complete. (Which is what happens on my systems where I have many snapshots.)

Whereas if they're wondering why they aren't getting all the properties they need, they'll look at the code.

Alternatively, if you only collected `-t filesystem` and `-t volume` (this would require two execs) and avoided snapshots by default, it would also -- potentially -- guard against submitting excessive data to the Chef Server.

**Comment by btm** [comment]
@bdha, I agree on being conservative but we only recently designed [RFC53](https://github.com/chef/chef-rfc/blob/master/rfc053-ohai-config.md) and implemented the ability to easily send configuration values to Ohai. Historically noisy Ohai plugins can move toward more conservative defaults and users that need more can be choosy via configuration.

Switching the filesystem plugin to be conservative on zfs properties for the next major release of Ohai would be a welcome patch.

**Comment by ksubrama** [comment]
:+1:

**Pull request merged**